### PR TITLE
GUI-2223 prevent stack_view from error-ing if cf-templates bucket doesn't exist

### PR DESCRIPTION
**Pull request by dkavanagh**
prevent stack_view from error-ing if cf-templates bucket doesn't exist

https://eucalyptus.atlassian.net/browse/GUI-2223

### CODE REVIEW

**Comment by euca-qa-gui-tester** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com/job/user-console-PR-monitor/1236/
Test FAILed.

**Comment by kamalgill** [comment]
Hal, retest this please

**Comment by euca-qa-gui-tester** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com/job/user-console-PR-monitor/1260/
Test FAILed.

**Comment by aliceh** [comment]
restest this, please

**Comment by aliceh** [comment]
Hal, retest this please

**Comment by dkavanagh** [comment]
restest this, please

**Comment by euca-qa-gui-tester** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com/job/user-console-PR-monitor/1266/
Test PASSed.

File: eucaconsole/views/stacks.py [hunk 1]
```diff
@@ -215,21 +215,25 @@ def __init__(self, request):
...
-            self.render_dict['template_name'] = name
-        else:
-            self.render_dict['template_name'] = None
+        self.render_dict['template_name'] = None
+        if bucket is not None:
+            keys = list(bucket.list(prefix=stack_hash))
+            if len(keys) > 0:
+                key = keys[0].key
+                name = key[key.rfind('-') + 1:]
+                self.render_dict['template_bucket'] = bucket.name
```
Comment (kamalgill) [inline eucaconsole/views/stacks.py]: @dkavanagh: Minor point, but lines 234-236 could be simplified via dict.update()...

```
self.render_dict.update({
    'template_bucket': bucket.name,
    'template_key': key,
    'template_name': name,
})
```

Comment (dkavanagh) [inline eucaconsole/views/stacks.py]: I do always wonder what's more pythonic, performant, clear to read.

**Pull request merged**